zhaquirks/tuya/ts0601_din_power.py

```diff
@@ -177,6 +183,78 @@ def _update_attribute(self, attrid, value):
             self.endpoint.electrical_measurement.power_factor_reported(value / 10)
 
 
+class ZemismartManufCluster(TuyaManufClusterAttributes):
+    """Manufacturer Specific Cluster of the Zemismart SPM01 Power Meter device."""
+
+    attributes = {
+        ZEMISMART_TOTAL_ENERGY_ATTR: ("energy", t.uint32_t, True),
+        ZEMISMART_TOTAL_REVERSE_ENERGY_ATTR: ("reverse_energy", t.uint32_t, True),
+        ZEMISMART_VCP_ATTR: ("vcp_raw", t.data64, True),
+    }
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == ZEMISMART_TOTAL_ENERGY_ATTR:
+            self.endpoint.smartenergy_metering.energy_deliver_reported(value)
+        elif attrid == ZEMISMART_TOTAL_REVERSE_ENERGY_ATTR:
+            self.endpoint.smartenergy_metering.energy_receive_reported(value)
+        elif attrid == ZEMISMART_VCP_ATTR:
+            self.endpoint.electrical_measurement.voltage_reported(
+                (value[7] * 256) + value[6]
+            )
+            self.endpoint.electrical_measurement.current_reported(
+                (value[5] * 256 * 256) + (value[4] * 256) + value[3]
+            )
+            self.endpoint.electrical_measurement.power_reported(
+                (value[2] * 256 * 256) + (value[1] * 256) + value[0]
+            )
+
+
+class ZemismartPowerMeasurement(LocalDataCluster, ElectricalMeasurement):
+    """Custom class for power, voltage and current measurement."""
+
+    cluster_id = ElectricalMeasurement.cluster_id
```

Comment:
This can be removed, as it's already set by inheriting from `ElectricalMeasurement`.

Reply:
You're right, will be fixed.
BTW, how should I implement three phase sensors? Which are the correct parameters for each phase in ElectricalMeasurement?

Reply:
Never mind, I think I've found out

Reply:
Hi, I own the "TS0601 by _TZE200_ves1ycwx" (three phase) and configured my hass for custom quirks yesterday.

![image](https://github.com/zigpy/zha-device-handlers/assets/28894450/d292dbe7-e248-42bb-88a7-22bb7812ebdf)

But only one entity is configured:
![image](https://github.com/zigpy/zha-device-handlers/assets/28894450/73852dc2-e98b-4e6e-b029-6890064524ad)

How can I help to get more entities ?

Reply:
@cnu80 Did you add the device before the quirk? If so, reconfigure the device and maybe restart HA. If that does not help, in the manage device menu check that vcp_raw from the ZemismartManufCluster is not None.